### PR TITLE
topology: topology1: Add initial topology for AMD Renoir platform

### PR DESCRIPTION
**Pull request by balakishorepati**
This PR adds m4 and headers to support basic topology with passthrough
pipeline for ACPSP and DMIC use case support on renoir platform.

### CODE REVIEW

File: tools/topology/topology1/CMakeLists.txt [hunk 1]
```diff
@@ -173,6 +173,7 @@ set(TPLGS
 	"sof-imx8mp-wm8960\;sof-imx8mp-drc-wm8960\;-DPPROC=drc"
 
 	"sof-mt8195-mt6359-rt1019-rt5682\;sof-mt8195-mt6359-rt1019-rt5682"
+	"sof-acp-renoir\;sof-acp"
```
Comment (plbossart) [inline tools/topology/topology1/CMakeLists.txt:176]: add newline above to clearly split Mediatek and AMD topologies please

File: tools/topology/topology1/sof-acp-renoir.m4 [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+# Include Token library
+include(`sof/tokens.m4')
+
+# Include ACP DSP configuration
+include(`platform/amd/acp.m4')
+
+# Playback pipeline 1 on PCM 0 using max 2 channels of s16le.
+# Schedule 96 frames per 2000us deadline on core 0 with priority 0
+PIPELINE_PCM_ADD(sof/pipe-passthrough-playback.m4,
```
Comment (plbossart) [inline tools/topology/topology1/sof-acp-renoir.m4:23]: I strongly suggest you use separate pipelines for host and dai parts, as done e.g. for nocodec and being added for HDaudio/SoundWire. It's more flexible. Using a single pipeline all the way from host to dai was a tragic mistake done many years ago. It doesn't scale.

Even if you only have one PCM device today, having a mixer by default will help you scale.

File: tools/topology/topology1/m4/dai.m4 [hunk 3]
```diff
@@ -155,7 +155,7 @@ define(`DO_DAI_CONFIG',
 `'
 `	id		"'$3`"'
 `'
-`	ifelse($1, `SSP', $5, $1, `HDA', $5, $1, `ALH', $5, $1, `ESAI', $5, $1, `SAI', $5, $1, `AFE', $5, `}')'
+`	ifelse($1, `SSP', $5, $1, `HDA', $5, $1, `ALH', $5, $1, `ESAI', $5, $1, `SAI', $5, $1, `AFE', $5, $1, `ACP', $5, $1, `ACPSP', $5, $1, `ACPDMIC', $5, `}')'
```
Comment (plbossart) [inline tools/topology/topology1/m4/dai.m4:158]: that line is starting to become unmaintainable. @ranj063 should we start cleaning this up?

**Review by plbossart — CHANGES_REQUESTED** [review]

File: tools/topology/topology1/m4/dai.m4 (hunk 3 above)
Comment (lgirdwood) [reply tools/topology/topology1/m4/dai.m4:158]: This should really be one of the last changes to M4 prior to deprecation.

Comment (ranj063) [reply tools/topology/topology1/m4/dai.m4:158]: wow. this check really needs to be cleaned up. The interesting part is we've covered almost all BE DAI types and all of them add the $5 argument even without bothering to check if there's even a 5th argument supplied. I will submit a PR

File: tools/topology/topology1/sof-acp-renoir.m4 (hunk 2 above)
Comment (bhiregoudar) [reply tools/topology/topology1/sof-acp-renoir.m4:23]: Thanks for the review and suggestions. This is the initial topology support. We will add more topology files and take care as suggested.

File: tools/topology/topology1/m4/dai.m4 (hunk 3 above)
Comment (lgirdwood) [reply tools/topology/topology1/m4/dai.m4:158]: @ranj063 do you have a PR number ?

File: tools/topology/topology1/CMakeLists.txt (hunk 1 above)
Comment (balakishorepati) [reply tools/topology/topology1/CMakeLists.txt:177]: Addressed the review comment.

File: tools/topology/topology1/m4/dai.m4 (hunk 3 above)
Comment (ranj063) [reply tools/topology/topology1/m4/dai.m4:158]: not yet @lgirdwood. will submit one tomorrow

**Review by ranj063 — APPROVED** [review]
looks good to me. It would be helpful to add the pipeline graph in the comments but not a showstopper

**Review by bhiregoudar — APPROVED** [review]

**Pull request merged**